src/main/java/me/iblitzkriegi/vixio/changers/EffChange.java

```diff
@@ -0,0 +1,344 @@
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.log.CountingLogHandler;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.skript.log.HandlerList;
+import ch.njol.skript.log.LogHandler;
+import ch.njol.skript.log.ParseLogHandler;
+import ch.njol.skript.log.SkriptLogger;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.Patterns;
+import ch.njol.skript.util.Utils;
+import ch.njol.util.Kleenean;
+import me.iblitzkriegi.vixio.Vixio;
+import me.iblitzkriegi.vixio.util.ReflectionUtils;
+import me.iblitzkriegi.vixio.util.Util;
+import me.iblitzkriegi.vixio.util.wrapper.Bot;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.logging.Level;
+
+/**
+ * @author Peter Güttinger
+ * edited minorly for Vixio
+ */
+public class EffChange extends Effect {
+
+    private static boolean parsing;
+
+    private static Patterns<ChangeMode> patterns = new Patterns<>(new Object[][]{
+            {"(add|give) %objects% to (%~objects%) with %bot/string%", ChangeMode.ADD},
+            {"increase %~objects% by (%objects%) with %bot/string%", ChangeMode.ADD},
+            {"give %~objects% (%objects%) as %bot/string%", ChangeMode.ADD},
+
+            {"set %~objects% to (%objects%) with %bot/string%", ChangeMode.SET},
+
+            {"remove (all|every) %objects% from (%~objects%) with %bot/string%", ChangeMode.REMOVE_ALL},
+
+            {"(remove|subtract) %objects% from (%~objects%) with %bot/string%", ChangeMode.REMOVE},
+            {"reduce %~objects% by (%objects%) with %bot/string%", ChangeMode.REMOVE},
+
+            {"(delete|clear) (%~objects%) with %bot/string%", ChangeMode.DELETE},
+
+            {"reset (%~objects%) with %bot/string%", ChangeMode.RESET}
+    });
+
+    static {
+        Vixio.getInstance().registerEffect(EffChange.class, patterns.getPatterns());
+    }
+
+    @SuppressWarnings("null")
+    private Expression<?> changed;
+    @Nullable
+    private Expression<?> changer = null;
+
+    private Expression<Bot> bot;
+
+    @SuppressWarnings("null")
+    private ChangeMode mode;
+
+    private boolean single;
+
+//	private Changer<?, ?> c = null;
+
+    @SuppressWarnings({"unchecked", "null"})
+    @Override
+    public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
+        parsing = true;
+        mode = patterns.getInfo(matchedPattern);
+        bot = (Expression<Bot>) exprs[2];
+
+        switch (mode) {
+            case ADD:
+                if (matchedPattern == 0) {
+                    changer = exprs[0];
+                    changed = exprs[1];
+                } else {
+                    changer = exprs[1];
```

Comment:
Shouldn't these two variables be the other way around, set exprs[0] then exprs[1]?

Reply:
no, they are correct